StoneMasonKarel.java

```diff
@@ -1,17 +1,85 @@
-/*
- * File: StoneMasonKarel.java
- * --------------------------
- * The StoneMasonKarel subclass as it appears here does nothing.
- * When you finish writing it, it should solve the "repair the quad"
- * problem from Assignment 1.  In addition to editing the program,
- * you should be sure to edit this comment so that it no longer
- * indicates that the program does nothing.
- */
-
-import stanford.karel.*;
-
-public class StoneMasonKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: StoneMasonKarel.java
+ * --------------------------
+ * The StoneMasonKarel subclass as it appears here does nothing.
+ * When you finish writing it, it should solve the "repair the quad"
+ * problem from Assignment 1.  In addition to editing the program,
+ * you should be sure to edit this comment so that it no longer
+ * indicates that the program does nothing.
+ */
+
+import stanford.karel.*;
+
+public class StoneMasonKarel extends SuperKarel {
+
+	 public void run() {
+		 turnLeft();
+		 moveOn();
+		 moveOn();
+		 moveOn();
+		 move();
+		 turnRight();
+		 moveFour();
+		 turnRight();
+		 moveOn();
+		 move();
+		 moveOn();
+		 move();
+		 
+		 turnLeft();
+		 moveFour();
+		 turnLeft();
+		 moveOn();
+		 moveOn();
+		 move();
+		 moveOn();
+		 turnRight();
+		 moveFour();
+		 
+		 turnRight();
+		 moveOn();
+		 moveOn();

```

Comment:
You need to do a better job at breaking down the problem in smaller sub-section and use those smaller sub-section to write your solution.